gadget/update.go

```diff
@@ -904,6 +905,247 @@ func buildNewVolumeToDeviceMapping(old GadgetData, laidOutVols map[string]*LaidO
 	}, nil
 }
 
+func buildVolumeStructureToLocation(old GadgetData,
+	isPreUC20 bool,
+	missingInitialMapping bool,
```

Comment:
have two bools here is not great, at a small extra cost passing model again instead of isPreUC20 would seem a bit clearer and missingInitialMapping could go at the end as anyway it refers to the contents of volToDeviceMapping

Reply:
sure I think you're right passing the model around is cleaner, I changed this